pkg/ottl/LANGUAGE.md

```diff
@@ -2,6 +2,15 @@
 
 The OTTL grammar includes function invocations, Values and Boolean Expressions. These parts all fit into a Statement, which is the basis of execution in the OTTL.
 
+### Design principles
+
+The OTTL is intended as a domain-specific language (DSL) for telemetry mutation and generation,
+and is not intended to be used as a general-purpose programming language.
+
+The OTTL has been designed to work directly with [pdata](https://github.com/open-telemetry/opentelemetry-collector/tree/main/pdata).
+While the OTTL can work with external data formats due to being implemented directly in Go instead of on top of a custom VM,
+it is not expressly intended to perform operations that don't touch `pdata` in some way.
```

Comment:
```suggestion
OTTL is intended as a domain-specific language (DSL) for telemetry mutation and generation,
and is not intended to be used as a general-purpose programming language.

OTTL has been designed to work directly with [pdata](https://github.com/open-telemetry/opentelemetry-collector/tree/main/pdata).
While OTTL can work with external data formats due to being implemented directly in Go instead of on top of a custom VM,
it is not expressly intended to perform operations that don't touch pdata in some way.
```

I'm sure there are inconsistencies elsewhere, but we made an editorial decision to drop the article from OTTL's name. Also, pdata is usually referred to without the literal quotes.